js/ad_manager_vast.js

```diff
@@ -935,41 +993,113 @@ OO.Ads.manager(function(_, $) {
+      }
+
+      var id = $(xml).attr("id");
+      if (typeof id !== 'undefined') {
+        result.id = id;
+      }
+
       return result;
     }, this);
 
     /**
-     * The xml needs to get parsed and and ad object is returned.
+     * The xml needs to get parsed and and an array of ad objects is returned.
      * @public
      * @method Vast#parser
      * @param {xml} vastXML The xml that contains the ad data
-     * @returns {object} If the ad is found it returns the object otherwise it returns null.
+     * @returns {object} If ads are found, an array containing the ad(s) is returned, otherwise it returns null.
      */
     this.parser = function(vastXML) {
-      if (!isValidVastXML(vastXML)) {
+      if (!this.isValidVastXML(vastXML)) {
         return null;
       }
+      var result = {
+        podded : [],
+        standalone : []
+      };
+      //parse the ad objects from the XML
+      var ads = this.parseAds(vastXML);
+      //check to see if any ads are sequenced (are podded)
+      _.each(ads, _.bind(function(ad) {
+        if (supportsPoddedAds(ad.version) && typeof ad.sequence !== 'undefined' && _.isNumber(parseInt(ad.sequence))) {
+          //Assume sequences will start from 1
+          result.podded[+ad.sequence - 1] = ad;
+        } else {
+          //store ad as a standalone ad
+          result.standalone.push(ad);
+        }
+      }, this));
 
-      var result = { ads: [] };
-      result.version = getVastVersion(vastXML);
-      var inline = $(vastXML).find("InLine");
-      var wrapper = $(vastXML).find("Wrapper");
+      return result;
+    };
 
-      if (inline.size() > 0) {
-        result.type = "inline";
-      } else if (wrapper.size() > 0) {
-        result.type = "wrapper";
-      } else {
-        return null;
-      }
+    /**
+     * Parses ad objects from the Vast XML.
+     * @public
+     * @method Vast#parseAds
+     * @param {xml} vastXML The xml that contains the ad data
+     * @return {Array} An array of ad objects
+     */
+    this.parseAds = function(vastXML) {
+      var result = [];
+      var version = getVastVersion(vastXML);
       $(vastXML).find("Ad").each(function() {
-        result.ads.push(VastAdSingleParser(this, result.type, result.version));
+        var singleAd = VastAdSingleParser(this, version);
+        if (singleAd) {
+          result.push(singleAd);
+        }
       });
-
       return result;
     };
 
     /**
+     * Prepares an array of ads to be added to the timeline, ready for playback.
+     * @private
+     * @method Vast#handleAds
+     * @param {Array} ads An array of ad objects
+     * @param {object} adLoaded The ad loaded object and metadata
+     */
+    var handleAds = _.bind(function(ads, adLoaded) {
+      //find out how many non linear ads we have so as to not count them
+      //when determining ad pod length
+      var nonLinearAdCount = 0;
+      _.each(ads, _.bind(function(ad) {
+        if (!_.isEmpty(ad.nonLinear)) {
```

Comment:
I think this will always evaluate to "not empty" because ad.nonLinear will always have "tracking" as a property like so:
```
ad.nonLinear = {
  tracking: {}
}
```
I think this happens because of LINE 789. double check me on that.